tardis/__init__.py

```diff
@@ -33,6 +33,114 @@
 logger.addHandler(console_handler)
 logging.getLogger("py.warnings").addHandler(console_handler)
 
+LOGGING_LEVELS = {
+    "NOTSET": logging.NOTSET,
+    "DEBUG": logging.DEBUG,
+    "INFO": logging.INFO,
+    "WARNING": logging.WARNING,
+    "ERROR": logging.ERROR,
+    "CRITICAL": logging.CRITICAL,
+}
+
+
+class FilterLog(object):
+    """
+    Filter Log Class for Filtering Logging Output
+    to a particular level
+
+    Parameters
+    ----------
+    log_level : logging object
+        allows to have a filter for the
+        particular log_level
+    """
+
+    def __init__(self, log_level):
+        self.log_level = log_level
+
+    def filter(self, log_record):
+        """
+        filter() allows to set the logging level for
+        all the record that are being parsed & hence remove those
+        which are not of the particular level
+
+        Parameters
+        ----------
+        log_record : logging.record
+            which the paricular record upon which the
+            filter will be applied
+
+        Returns
+        -------
+        boolean : True, if the current log_record has the
+            level that of the specified log_level
+            False, if the current log_record doesn't have the
+            same log_level as the specified one
+        """
+        return log_record.levelno == self.log_level
+
+
+# Setting up a list to store the Logging Filters set by logger.addFilter()
+list_of_filter = []
+
+
+def logging_state(log_state, tardis_config, specific):
+    """
+    Function to set the logging configuration for the simulation output
+    Called from within run_tardis()
+    Configured via functional arguments passed through run_tardis() - log_state & specific
+    Configured via YAML parameters under `debug` section - logging_level & specific_logging
+
+    Parameters
+    ----------
+    log_state: str
+        Allows to input the log level for the simulation
+        Uses Python logging framework to determine the messages that will be output
+    specific: boolean
+        Allows to set specific logging levels. Logs of the `log_state` level would be output.
+    """
+
+    if "debug" in tardis_config:
+        specific = (
+            tardis_config["debug"]["specific"] if specific is None else specific
+        )
+
+        logging_level = (
+            log_state if log_state else tardis_config["debug"]["log_state"]
+        )
+        logging_level = logging_level.upper()
+
+        if not logging_level in LOGGING_LEVELS:
+            raise ValueError(
+                f"Passed Value for log_state = {logging_level} is Invalid. Must be one of the following {list(LOGGING_LEVELS.keys())}"
+            )
+    else:
+        tardis_config["debug"] = {"log_state": "CRITICAL"}
```

Comment:
```suggestion
        tardis_config["debug"]["log_state"] = "CRITICAL"
```

Reply:
Sure, and **CRITICAL** will be made into a global constant value such as `DEFAULT_LOG_STATE` such that it makes more sense why this value has been assigned 😉

Reply:
Resolved 😄

Reply:
I think so this cannot be done. It is failing the tests again 🤔

Reply:
There is a problem with this kind of assignment. What is happening is that when we are assigning the value through `tardis_config["debug"]["log_state"] = "CRITICAL"`, it actually checks for the key `debug` & then for `log_state` for assignment. But if the keys are not already present then, it raises `KeyError`. 
In my implementation, I have actually first added the `debug` key then through a dictionary added the value to the `log_state` parameter. Hence, when assigning the value in the subsequent line, the key is actually present & no error is raised.